src/posthog-core.ts

```diff
@@ -740,9 +740,11 @@ export class PostHog {
 
         if (this.config.store_google) {
             this.sessionPersistence.update_campaign_params()
+            this.persistence.set_initial_campaign_params()
         }
         if (this.config.save_referrer) {
             this.sessionPersistence.update_referrer_info()
+            this.persistence.set_initial_referrer_info()
         }
```

Comment:
is the coupling "hard" here... i.e. if you call `this.sessionPersistence.update_campaign_params()` should you always call `this.persistence.set_initial_campaign_params()` or vice versa?

we could make that explicit so that callers don't have to remember... these two persistence stores by necessity will be super coupled anyway

but I'm thinking out loud - feel free to ignore

Reply:
To *actually* answer your question which I kinda avoided :D

While the caller is "us", I'm not too worried that we have to remember which persistence to us. The alternative would be to have a combined persistence, where the caller just says what it want to save and it calls the right storage. That would be doable but I'd worry about increasing the bundle size by doing so.